src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java

```diff
@@ -86,7 +92,24 @@ public int join() throws IOException, InterruptedException {
             LOGGER.log(Level.FINEST, "Waiting for websocket to close on command finish ({0})", finished);
             finished.await();
             LOGGER.log(Level.FINEST, "Command is finished ({0})", finished);
-            return watch.exitCode().join();
+
+            CompletableFuture<Integer> exitCodeFuture = watch.exitCode();
+            Integer exitCode = exitCodeFuture.get();
```

Comment:
Does this not contradict the PR title, that it is possible `exitCodeFuture == null`? Either the code or the PR title is incorrect.

Reply:
(Docs are ambiguous whether `null` refers to `CompletableFuture<Integer>`, or the value of the `CompletableFuture`.)

Reply:
The goal here is to handle all possible outcomes (null `exitCode` is handled in this block, null `exitCodeFuture` in catchall block, checked `ExecutionException` that can be thrown by `get() ` in catch block). 
The PR title refers to the original issue that's causing NPEs on current master when `exitCode` is null, but what would you suggest as a better naming/solution here?

Reply:
> null `exitCodeFuture` in catchall block

If you mean in the `catch (Exception` block, then yes, but code should never knowingly be catching `NullPointerException`; it should check for null. Conversely, if you were relying on arbitrary unknown exceptions, you might as well keep it simple

```java
try {
    return watch.exitCode().join();
} catch (Exception x) {
    LOGGER.log(Level.FINEST, "Exception occurred while waiting for exit code", x);
    return -1;
}
```

Reply:
Alright, agreed - how about now?

Reply:
@jglick could you please re-review?